Automattic/jetpack
# Debug Helper IDC Simulator: add a button to trigger a remote request

#### Changes proposed in this Pull Request:
* Remote requests need to be sent to WPCOM  when testing the IDC process. To make testing easier, add a button to the IDC Simulator page which triggers an authenticated remote request to WPCOM.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install, activate, and connect Jetpack.
2. Install and activate this branch of the Debug Helper plugin.
3. Enable the Debug Helper plugin's IDC Simulator module.
4. To make testing easier, set the `JETPACK__SANDBOX_DOMAIN` constant to your sandbox. This will cause every remote request to be logged in the `debug.log` file.
5. On the IDC Simulator page, click the `Send Remote Request` button. Check your site's `debug.log` file and verify that a request was sent to `https://public-api.wordpress.com/wpcom/v2/sites/182848265/jetpack-token-health/blog`.
6. On the IDC Simulator page, verify that the `The remote request was successfully sent!` admin notice is displayed.
7. To make sure that the admin notices still display correctly, click the `Store these options` button on the IDC Simulator page.
8. Verify that the `IDC simulation settings have been saved!` admin notice is displayed.
